include/rocksdb/db.h

```diff
@@ -280,9 +281,24 @@ class DB {
   // a status for which Status::IsNotFound() returns true.
   //
   // May return some other Status on an error.
+  inline Status Get(const ReadOptions& options,
+                    ColumnFamilyHandle* column_family, const Slice& key,
+                    std::string* value) {
+    if (LIKELY(value != nullptr)) {
+      PinnableSlice pinnable_val;
+      auto s = Get(options, column_family, key, &pinnable_val);
```

Comment:
If I understand correctly, does that mean that we introduce an extra memcpy for memtable/merge operator get ?
```
memtable -> PinnableSlice::self_space -> value
```

Let's try 
```
std::move
```
or
We can allow PinnableSlice to accept an external space
```
PinnableSlice {
std::string* own_data_ptr_; // This point to own_data_ except if we change it to something else
std::string own_data_;
}
```


We can measure the regression by running db_bench and making sure that all keys live in memtable
```
./db_bench --benchmarks="fillseq,stats,readrandom" --num=<something_small_enough>

we can verify that all the keys are in the memtable by looking at the stats result and see that there are no files generated in L0 or any other levels
```